# HG changeset patch
# User Rik <rik@octave.org>
# Date 1317226485 25200
#      Wed Sep 28 09:14:45 2011 -0700
# Node ID 41c2f4633a621be8cad514cd3b769d21e83c5b67
# Parent  dd3c5325039cff44d62191e364b137ca3f7eb1e7
Properly process 'cdata' property for patches (Bug #34417).

* __patch__.m: Guarantee that 'cdata' and 'facevertexcdata' are transposes of
each other.

diff --git a/scripts/plot/private/__patch__.m b/scripts/plot/private/__patch__.m
--- a/scripts/plot/private/__patch__.m
+++ b/scripts/plot/private/__patch__.m
@@ -284,17 +284,17 @@ function args = setvertexdata (args)
     vert = [x(:), y(:)];
   endif
   faces = reshape (1:numel(x), rows (x), columns (x));
   faces = faces';
 
   if (ndims (c) == 3)
     fvc = reshape (c, size (c, 1) * size (c, 2), size(c, 3));
   else
-    fvc = c(:).';
+    fvc = c.';
   endif
 
   args = {"faces", faces, "vertices", vert, "facevertexcdata", fvc, args{:}};
 endfunction
 
 function update_data (h, d)
   update_handle (h, false);
 endfunction
